keycloak/sms-provider/src/main/java/org/sunbird/keycloak/login/phone/AbstractPhoneFormAuthenticator.java

```diff
@@ -82,11 +82,14 @@ public boolean validateUserAndPassword(AuthenticationFlowContext context,
       return false;
     }
 
-    if (invalidUser(context, user)) {
+    /*if (invalidUser(context, user)) {
       return false;
-    }
+    }*/
+    //found the method name but we need to see how error will be responded back
+    testInvalidUser(context, user);
 
-    if (!validatePassword(context, user, inputData)) {
+    //based on  innner method implementation clearUser value set as true
+    if (!validatePassword(context, user, inputData, true)) {
```

Comment:
validatePassword: New parameter "clearUser" got added in exceptional scenario it will be helpful to removeing user details from the current session, else user details still will be there in context.